akka-docs-dev/rst/java/http/client-side/host-level.rst

```diff
@@ -1,8 +1,147 @@
-.. _HostLevelApi-java:
+.. _host-level-api-java:
 
 Host-Level Client-Side API
 ==========================
 
-TODO
+As opposed to the :ref:`connection-level-api` the host-level API relieves you from manually managing individual HTTP
+connections. It autonomously manages a configurable pool of connections to *one particular target endpoint* (i.e.
+host/port combination).
 
-For the time being, :ref:`see the Scala chapter on the same topic <HostLevelApi>`.
\ No newline at end of file
+
+Requesting a Host Connection Pool
+---------------------------------
+
+The best way to get a hold of a connection pool to a given target endpoint is the ``Http.get(system).cachedHostConnectionPool(...)``
+method, which returns a ``Flow`` that can be "baked" into an application-level stream setup. This flow is also called
+a "pool client flow".
+
+The connection pool underlying a pool client flow is cached. For every ``ActorSystem``, target endpoint and pool
+configuration there will never be more than a single pool live at any time.
+
+Also, the HTTP layer transparently manages idle shutdown and restarting of connection pools as configured.
+The client flow instances therefore remain valid throughout the lifetime of the application, i.e. they can be
+materialized as often as required and the time between individual materialization is of no importance.
+
+When you request a pool client flow with ``Http.get(system).cachedHostConnectionPool(...)`` Akka HTTP will immediately start
+the pool, even before the first client flow materialization. However, this running pool will not actually open the
+first connection to the target endpoint until the first request has arrived.
+
+
+Configuring a Host Connection Pool
+----------------------------------
+
+Apart from the connection-level config settings and socket options there are a number of settings that allow you to
+influence the behavior of the connection pool logic itself.
+Check out the ``akka.http.client.host-connection-pool`` section of the Akka HTTP :ref:`akka-http-configuration-java` for
+more information about which settings are available and what they mean.
+
+Note that, if you request pools with different configurations for the same target host you will get *independent* pools.
+This means that, in total, your application might open more concurrent HTTP connections to the target endpoint than any
+of the individual pool's ``max-connections`` settings allow!
+
+There is one setting that likely deserves a bit deeper explanation: ``max-open-requests``.
+This setting limits the maximum number of requests that can be open at any time for a single connection pool.
```

Comment:
what does it mean when a request is "open"? i think that term rather applies to connections. maybe "in flight"?